### ISSUE TITLE
Add value-no-empty rule

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#4312]
> What is the problem you're trying to solve?

Lately I've been noticing on website properties without values. Browser ignores them, but it could be a possible error.

> What solution would you like to see?

 New rule to avoid missing values. The following patterns are considered violations:

```css
a {
    color: ;
}

a {
    color:;
}

a {
    color: /* comment */;
}
```

I'm not sure about the name. Should it be `value-no-empty` or `declaration-property-value-no-empty`?

I would like to implement this rule myself.

**Comment by evilebottnawi** [comment]
It is valid css syntax?

**Comment by hudochenkov** [comment]
No.

**Comment by evilebottnawi** [comment]
@hudochenkov hm, i think it is out of scope stylelint, maybe in future we can create the rule no-invalid-syntax or something similar, but now better avoid this, my 2 cents

**Comment by hudochenkov** [comment]
I think it could fit styelint. https://stylelint.io/vision#possible-errors

**Comment by evilebottnawi** [comment]
@hudochenkov if we create rule we need other cases like `selector-no-empty`, `property-no-empty`, `media-feature-name-no-empty` and etc, it is out of scope stylelint, because it is syntax problems, better create plugin for this purpose

**Comment by vankop** [comment]
@hudochenkov  @evilebottnawi Looks like you are both right 🙈 

I, personally, agree with @evilebottnawi suggestion

**Comment by ntwb** [comment]
Does the `stylelint-csstree-validator` plugin detect missing values, I think it does and this is why we removed these types of rules from stylelint a while ago.

**Comment by jeddy3** [comment]
> Does the `stylelint-csstree-validator` plugin detect missing values, I think it does and this is why we removed these types of rules from stylelint a while ago.

Yes it does.

> I think it does and this is why we removed these types of rules from stylelint a while ago.

That's correct. It was `selector-no-empty` [we removed in 8.0.0](https://github.com/stylelint/stylelint/blob/master/CHANGELOG.md#800).

> if we create rule we need other cases like selector-no-empty, property-no-empty, media-feature-name-no-empty and etc, it is out of scope stylelint, because it is syntax problems, __better create plugin for this purpose__

I think I agree. If `stylelint-csstree-validator` isn't currently appropriate for this use case then a `stylelint-validate` plugin pack is definitely an interim option. 

Can I close this issue?

**Comment by Mouvedia** [comment]
> Can I close this issue?

Yup.
